### PR TITLE
Fix AggregateTypeResolver to respect interface, improve test coverage

### PR DESCRIPTION
**Pull request by ben-challis**

### CODE REVIEW

File: tests/unit/AggregateVersionTest.php [hunk 1]
```diff
@@ -49,7 +49,9 @@ public function it_throws_when_constructing_from_integer_less_than_one(int $less
      */
     public function it_can_provide_the_next_aggregate_version_that_follows_it(): void
     {
-        $this->assertSame(2, AggregateVersion::fromInteger(1)->next()->toInteger());
+        $v1 = AggregateVersion::fromInteger(1);
+        $this->assertSame(2, $v1->next()->toInteger());
+        $this->assertSame(1, $v1->toInteger());
```
Comment (ben-challis) [inline tests/unit/AggregateVersionTest.php:54]: Verifies lack of side-effects on original instance.

File: src/MapAggregateTypeResolver.php [hunk 2]
```diff
@@ -34,9 +34,10 @@ public function __construct(array $map)
...
     public function resolve(object $aggregate): AggregateType
     {
-        $aggregateType = $this->map[$aggregate::class] ?? null;
-        \assert($aggregateType instanceof AggregateType);
+        if (!isset($this->map[$aggregate::class])) {
+            throw CannotResolveAggregateType::of($aggregate, 'Class is not mapped');
+        }
 
-        return $aggregateType;
+        return $this->map[$aggregate::class];
```
Comment (marmichalski) [inline src/MapAggregateTypeResolver.php:41]:
```
return $this->map[$aggregate::class] ?? throw CannotResolveAggregateType::of($aggregate, 'Class is not mapped');
```

Comment (ben-challis) [reply src/MapAggregateTypeResolver.php]: Fixed in latest (force push).

File: tests/unit/ClosureAggregateIdExtractorTest.php [hunk 3]
```diff
@@ -36,4 +37,45 @@ protected function createAggregateWithExpectedAggregateId(): object
...
+
+    /**
+     * @test
+     */
+    public function it_rethrows_any_interface_compliant_exception_from_the_closure(): void
+    {
+        $aggregate = new class () {
+        };
+        $exception = CannotExtractAggregateId::from($aggregate);
+        $this->expectExceptionObject($exception);
```
Comment (marmichalski) [inline tests/unit/ClosureAggregateIdExtractorTest.php:49]: nit: I'd personally move expectancy close to the place where it's expected, that is above extract call, not a big deal tho

**Review by marmichalski — APPROVED** [review]

File: tests/unit/ClosureAggregateIdExtractorTest.php (hunk 3 above)
Comment (ben-challis) [reply tests/unit/ClosureAggregateIdExtractorTest.php]: Fixed in https://github.com/Lendable/aggregate/pull/308/commits/d21387426bc6ed5768477d467ea881a702f5385d

**Review by marmichalski — APPROVED** [review]

**Comment by ben-challis** [comment]
Thanks @marmichalski

**Pull request merged**